### PR TITLE
Support for mongodb+srv url

### PR DESCRIPTION
**Pull request by rafaelm7o**
This is the implementation for supporting mongodb+srv URL
I tested it with a MongoDB Atlas Account-
For the test cases, I simulated the SRV and TXT record with my own domain

### CODE REVIEW

File: lib/mongo/url_parser.ex [hunk 1]
```diff
@@ -114,6 +114,7 @@ defmodule Mongo.UrlParser do
     case :os.type() do
       {:win32, _} ->
         :inet_db.add_ns({4, 2, 2, 1})
+      _ ->
```
Comment (ankhers) [inline lib/mongo/url_parser.ex]: I would rather return an explicit value here instead of the implicit `nil`.

**Comment by ankhers** [comment]
I would also rather not rely on your domain for the tests. Maybe we should look into what official drivers are doing for their tests?

**Comment by rafaelm7o** [comment]
The problem was I did not find any test official domain for that, That is the reason for I decided to use the custom domain

**Comment by ankhers** [comment]
* [ruby driver](https://github.com/mongodb/mongo-ruby-driver/search?utf8=%E2%9C%93&q=srv&type=)
* [python driver](https://github.com/mongodb/mongo-python-driver/search?q=srv&unscoped_q=srv)

It looks like there is a number of 10gen.cc domains that they are using.

**Comment by rafaelm7o** [comment]
I changed the URL. I haven't seen the test cases, and certainly, some will fail. 
Ex. with more than one TXT record.

It works in a happy path scenery

**Comment by ankhers** [comment]
Thank you!

**Pull request merged**